common/beerocks/bcl/source/beerocks_thread_base.cpp

```diff
@@ -38,7 +38,7 @@ void thread_base::join()
 void thread_base::stop(bool block)
 {
     should_stop = true;
-    before_stop();
+    thread_base::before_stop();
```

Comment:
I don't think this is correct... This is declared empty in thread_base, so if the intention was to call the one from thread_base, it would just have been left out.

But indeed, the code is in fact broken, because when called from the destructor, the derived class's before_stop is no longer available!

So, I think the proper fix is to:
- duplicate the code below to the destructor instead of calling stop();
- add a destructor to async_work_queue that calls its own before_stop().

This is not a nitpick!

Reply:
Is this resolved?

Reply:
Yes, done. Nice catch BTW!